tests/cases/compiler/commonJsIsolatedModules.ts

```diff
@@ -0,0 +1,12 @@
+// @Filename: tsconfig.json
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "noEmit": true,
```

Comment:
What happens when `noEmit` is `false`?

Reply:
error saying that you can't overwrite index.js. The real example of this is from babel pipelines, where we recommended people use isolatedModules, so they use babel for emit, not typescript.

Reply:
But you could change it to emit into the `outDir` right? Would it affect in some way the emit that could happen?

Reply:
I'll try it.

Reply:
It shouldn't affect emit because we're not transforming js as far as I know. I pushed a commit with `outDir` replacing `noEmit` in the test.